### PR TITLE
HF: Add hf-tree-creator-lc-to-k0s-p

### PR DESCRIPTION
**Pull request by DanielSamitz**
Added the `hf-tree-creator-lc-to-k0s-p` to config_task.sh and dpl-config_run3.json. It was already included in workflows.yml.

### CODE REVIEW

**Comment by DanielSamitz** [comment]
In dpl-config_run3.json I set `"processMc": "true"` and `"processData": "false"` because that is the default. I noticed that here the other tree creators have both process switches set to `"true"`? Does this make sense in this context?

**Comment by vkucera** [comment]
Thanks @DanielSamitz . The process data/MC switches are overwritten depending on the data set so that should be fine.
The table names in `workflows.yml` seem incorrect. Please fix them and test that they are written out when `SAVETREES=1`.

**Comment by vkucera** [comment]
Also please add `DOO2_TREE_LCK0SP` in the condition for [`SUFFIX_SKIM`](https://github.com/AliceO2Group/Run3Analysisvalidation/blob/master/codeHF/config_tasks.sh#L342C87-L342C98).

**Comment by DanielSamitz** [comment]
Hi @vkucera , a question about the AdjustJSON. I have to switch `processCascades` to true and `processNoCascades` to false for the skim creator. What should be the condition for this?
Should it be 
`[[ $DOO2_CAND_CASC -eq 1 || $DOO2_SEL_LCK0SP -eq 1 || $DOO2_TASK_LCK0SP -eq 1 || $DOO2_TREE_LCK0SP ]]` (The same condition as for `SUFFIX_SKIM`)?

**Comment by vkucera** [comment]
> Hi @vkucera , a question about the AdjustJSON. I have to switch `processCascades` to true and `processNoCascades` to false for the skim creator. What should be the condition for this? Should it be `[[ $DOO2_CAND_CASC -eq 1 || $DOO2_SEL_LCK0SP -eq 1 || $DOO2_TASK_LCK0SP -eq 1 || $DOO2_TREE_LCK0SP ]]` (The same condition as for `SUFFIX_SKIM`)?

Yes, that should be fine. You should test it in any case.

**Comment by DanielSamitz** [comment]
Hi @vkucera , sorry for the long delay. I included the changes in SUFFIX_SKIM and AdjustJson, and changed the names of the tables in workflows.yml. I tested that the tree-creator-lc-to-k0s-p workflow works and the trees are written correctly.

Please note that in AdjustJson I also needed to include a swapping for `processData` true<-->false, depending on whether `INPUT_IS_MC `is selected or not. This affects of course also other tree creators, but I think they need this as well. Because that switch was not set properly when changing between MC/Data.

In the dpl-config.json I needed to change some options (`useWeightedFinalPCA` to false in track-index-skim-creator-cascades and candidate-creator-cascades and `tpcRefitBach` and `tpcRefitV0Daugh` to false in track-index-skim-creator-cascades) to make the workflow run without errors or empty histograms. This is now not consistent with the defaults of the configurables in the code, but is what we are currently always using to make it run.

**Comment by DanielSamitz** [comment]
I included now also the changes related to the most recent update for the tree-creator here https://github.com/AliceO2Group/O2Physics/pull/3615

File: codeHF/config_tasks.sh [hunk 1]
```diff
@@ -187,11 +188,13 @@ function AdjustJson {
     ReplaceString "\"processMc\": \"false\"" "\"processMc\": \"true\"" "$JSON" || ErrExit "Failed to edit $JSON."
     ReplaceString "\"processMC\": \"false\"" "\"processMC\": \"true\"" "$JSON" || ErrExit "Failed to edit $JSON."
     ReplaceString "\"isMC\": \"false\"" "\"isMC\": \"true\"" "$JSON" || ErrExit "Failed to edit $JSON."
+    ReplaceString "\"processData\": \"true\"" "\"processData\": \"false\"" "$JSON" || ErrExit "Failed to edit $JSON."
```
Comment (vkucera) [inline codeHF/config_tasks.sh:191]: I am not sure this is correct. Some workflows run `processData` also in MC.

File: codeHF/config_tasks.sh [hunk 2]
```diff
@@ -215,6 +218,13 @@ function AdjustJson {
     ReplaceString "\"doCutQuality\": \"true\"" "\"doCutQuality\": \"false\"" "$JSON" || ErrExit "Failed to edit $JSON."
   fi
 
+  # hf-track-index-skim-creator-cascades
+  if [[ $DOO2_CAND_CASC -eq 1 || $DOO2_SEL_LCK0SP -eq 1 || $DOO2_TASK_LCK0SP -eq 1 || $DOO2_TREE_LCK0SP -eq 1 ]]; then
+    ReplaceString "\"processCascades\": \"false\"" "\"processCascades\": \"true\"" "$JSON" || ErrExit "Failed to edit $JSON."
+    ReplaceString "\"processNoCascades\": \"true\"" "\"processNoCascades\": \"false\"" "$JSON" || ErrExit "Failed to edit $JSON."
+  fi
+
+
```
Comment (vkucera) [inline codeHF/config_tasks.sh:227]:
```suggestion
```

File: codeHF/config_tasks.sh (hunk 1 above)
Comment (DanielSamitz) [reply codeHF/config_tasks.sh:191]: OK, then my proposed solution here is of course problematic. But at least some tree creators (including the lc-to-k0s-p one), need to run exclusively `processData` or `processMc`, so we need to switch that on/off.

Comment (vkucera) [reply codeHF/config_tasks.sh:191]: Then those devices should have `processData` set to `false` by default in the JSON.

Comment (DanielSamitz) [reply codeHF/config_tasks.sh:191]: Ok, for the tree-creator-lc-to-k0s-p I have it already like this, for the tree-creator-lc-to-p-k-pi `processData` is currently set to `true` in the dpl-config_Run3.json. Should I change that too?
Did I get you correctly that then I should only remove the line
`ReplaceString "\"processData\": \"true\"" "\"processData\": \"false\"" "$JSON" || ErrExit "Failed to edit $JSON."`
and leave the opposite 
`ReplaceString "\"processData\": \"false\"" "\"processData\": \"true\"" "$JSON" || ErrExit "Failed to edit $JSON."`
in as it is now?

**Review by vkucera — APPROVED** [review]
Thanks @DanielSamitz

**Pull request merged**